src/document.h

```diff
@@ -1175,12 +1175,11 @@ struct Document {
                 return nullptr;
             }
 
-            case A_SEARCHNEXT: {
-                return SearchNext(dc, false, true, false);
-            }
-
+            case A_SEARCHNEXT: 
             case A_SEARCHPREV: {
-                return SearchNext(dc, false, true, true);
+                return sys->searchstring.Len()
+                    ? SearchNext(dc, false, true, k == A_SEARCHPREV)
+                    : LoadSearch();

```

Comment:
this function only sets the search, so does that mean I have to press search twice to do the actual search? what is the idea behind that UI (as opposed to searching directly?)

Reply:
I think of two reasons:
1. First provide some kind of visual feedback that the search is now active (the cells are darkened) and ready to proceed.
2. The search parameters are set by `MyFrame::OnSearch` that is run when the value of the `wxTextCtrl` `filter` is changed. So this runs asynchronously and we cannot just run `SearchNext` without knowing for sure that the search parameters are all set.

Reply:
Ok, I suppose that's fine for the moment.. much better than not working at all :)

Reply:
Thanks for your review!